src/m365/spo/commands/commandset/commandset-set.ts

```diff
@@ -0,0 +1,235 @@
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Updating ListView Command Set ${args.options.clientSideComponentId} to site '${args.options.webUrl}'...`);
+    }
+
+    if (!args.options.scope) {
+      args.options.scope = 'Site';
+    }
+
+    const location: string = this.getLocation(args.options.location ? args.options.location : '');
+
+    try {
+      const requestBody: any = {};
+
+      if (args.options.newTitle) {
+        requestBody.Title = args.options.newTitle;
+      }
+
+      if (args.options.location) {
+        requestBody.Location = location;
+      }
+
+      if (args.options.listType) {
+        requestBody.RegistrationId = this.getListTemplate(args.options.listType);
+      }
+
+      if (args.options.clientSideComponentProperties) {
+        requestBody.ClientSideComponentProperties = args.options.clientSideComponentProperties;
+      }
+
+      const id = await this.getCommandSetId(args.options);
+
+      const requestOptions: any = {
+        url: `${args.options.webUrl}/_api/${args.options.scope}/UserCustomActions('${formatting.encodeQueryParameter(id)}')`,
+        headers: {
+          accept: 'application/json;odata=nometadata',
+          'X-HTTP-Method': 'MERGE'
+        },
+        data: requestBody,
+        responseType: 'json'
+      };
+
+      await request.post<CustomAction>(requestOptions);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getCommandSetId(options: Options): Promise<string> {
+    if (options.id) {
+      return options.id;
+    }
+
+    if (options.title) {
+      const commandSets: CustomAction[] = await spo.getCustomActions(options.webUrl, options.scope, `(Title eq '${formatting.encodeQueryParameter(options.title as string)}') and (startswith(Location,'ClientSideExtension.ListViewCommandSet'))`);
+
+      if (commandSets.length === 0) {
+        throw `No user commandsets with title '${options.title}' found`;
+      }
+
+      if (commandSets.length > 1) {
+        throw `Multiple user commandsets with title '${options.title}' found. Please disambiguate using IDs: ${commandSets.map(a => a.Id).join(', ')}`;
+      }
+
+      return commandSets[0].Id;
+    }
+
+    const commandSets: CustomAction[] = await spo.getCustomActions(options.webUrl, options.scope, `(ClientSideComponentId eq guid'${options.clientSideComponentId}') and (startswith(Location,'ClientSideExtension.ListViewCommandSet'))`);
+
+    if (commandSets.length === 0) {
+      throw `No user commandsets with ClientSideComponentId '${options.clientSideComponentId}' found`;
+    }
+    if (commandSets.length > 1) {
+      throw `Multiple user commandsets with ClientSideComponentId '${options.clientSideComponentId}' found. Please disambiguate using IDs: ${commandSets.map((commandSet: CustomAction) => commandSet.Id).join(', ')}`;
+    }
+    return commandSets[0].Id;
```

Comment:
I just noticed that now when we removed the `if` blocks that we may remove bunch of lines and combine a bit the code to something like this

```
let commandSets: CustomAction[] = [];
    if (options.title) {
      commandSets = await spo.getCustomActions(options.webUrl, options.scope, `(Title eq '${formatting.encodeQueryParameter(options.title as string)}') and (startswith(Location,'ClientSideExtension.ListViewCommandSet'))`);
    }
    else {
      commandSets = await spo.getCustomActions(options.webUrl, options.scope, `(ClientSideComponentId eq guid'${options.clientSideComponentId}') and (startswith(Location,'ClientSideExtension.ListViewCommandSet'))`);
    }

    if (commandSets.length === 0) {
      throw `No user commandsets with ${options.title ? `title '${options.title}'` : `ClientSideComponentId '${options.clientSideComponentId}'`} found`;
    }

    if (commandSets.length > 1) {
      throw `Multiple user commandsets with ${options.title ? `title '${options.title}'` : `ClientSideComponentId '${options.clientSideComponentId}'`} found. Please disambiguate using IDs: ${commandSets.map((commandSet: CustomAction) => commandSet.Id).join(', ')}`;
    }

    return commandSets[0].Id;
```

could you give it a double check? In The end we will produce code which is easier to maintain